.pre-commit-config.yaml

```
@@ -133,14 +133,14 @@ repos:
     hooks:
       - id: htmlhint
   - repo: https://github.com/igorshubovych/markdownlint-cli
-    rev: v0.39.0
+    rev: v0.40.0
     hooks:
       - id: markdownlint
         args: [--disable, MD013]
       - id: markdownlint-fix
         args: [--disable, MD013]
   - repo: https://github.com/pdm-project/pdm
-    rev: 2.15.0
+    rev: 2.15.3
```

@ritwik-g we use [PDM version 2.12.3](https://github.com/Zipstack/unstract/blob/798b5c96fa99c3c7c046f654773d151e9f01729f/docker/dockerfiles/backend.Dockerfile#L14) in our Dockerfiles and dev envs - it might be better to pin the lock check to the same version as well.
Or if there's nothing blocking us from upgrading - we should upgrade PDM to 2.15.3.
CC: @hari-kuriakose @nehabagdia 
```suggestion
    rev: 2.12.3
```

---

@chandrasekharan-zipstack good point. But already we are using `2.15.0` in the pre-commit and it have worked so far. So I don't think we should roll back now. We can plan upgrading docker pdm versions.

---

We can upgrade PDM versions in Dockerfiles and dev env later. Should be ok to proceed now.